main/pipeline/pipeline.cc

```diff
@@ -288,26 +265,75 @@ parser::ParseResult runPrismParser(core::GlobalState &gs, core::FileRef file, co
         core::UnfreezeNameTable nameTableAccess(gs); // enters strings from source code as names
         // The RBS rewriter produces plain Whitequark nodes and not `NodeWithExpr` which causes errors in
         // `PrismDesugar.cc`. For now, disable all direct translation, and fallback to `Desugar.cc`.
-        auto directlyTranslate = !gs.cacheSensitiveOptions.rbsEnabled;
-        parseResult = parser::Prism::Parser::run(ctx, directlyTranslate);
+        auto source = file.data(ctx).source();
+        parser::Prism::Parser parser{source};
+        bool collectComments = gs.cacheSensitiveOptions.rbsEnabled;
+        parser::Prism::ParseResult prismResult = parser.parseWithoutTranslation(collectComments);
+
+        auto node = prismResult.getRawNodePointer();
+        // TODO: Enable once RBS Prism pipeline is fully implemented. For now, we use the legacy RBS rewrite
+        // path (see indexOne): https://github.com/sorbet/sorbet/issues/9065
+        // if (gs.cacheSensitiveOptions.rbsEnabled) {
+        //     node = runPrismRBSRewrite(gs, file, node, prismResult.getCommentLocations(), print, ctx, parser);
+        // }
+
+        bool directlyDesugar = !gs.cacheSensitiveOptions.rbsEnabled;
+        auto translatedTree = parser::Prism::Translator(parser, ctx, prismResult.getParseErrors(), directlyDesugar,
+                                                        preserveConcreteSyntax)
+                                  .translate(node);
+
+        parseResult = parser::ParseResult{move(translatedTree), prismResult.getCommentLocations()};
     }
 
     if (print.ParseTree.enabled) {
-        print.ParseTree.fmt("{}\n", parseResult.tree->toStringWithTabs(gs, 0));
+        if (parseResult.tree) {
```

Comment:
Do we need these checks more than before? Previous implementation didn't have them. We should either combine all the checks and return `parseResult` early or not check at all.

Reply:
Let's move it to a single check before all the print `if` statements.